src/infra/repositories/indexer_repository.rs

```diff
@@ -50,73 +52,211 @@ pub struct UpdateIndexerStatusAndProcessIdDb {
     pub process_id: i64,
 }
 
-pub async fn _insert(pool: &Pool<AsyncPgConnection>, new_indexer: NewIndexerDb) -> Result<IndexerModel, InfraError> {
+#[async_trait]
+pub trait Repository {
+    async fn insert(&mut self, new_indexer: NewIndexerDb) -> Result<IndexerModel, InfraError>;
+    async fn get(&self, id: Uuid) -> Result<IndexerModel, InfraError>;
+    async fn get_all(&self, filter: IndexerFilter) -> Result<Vec<IndexerModel>, InfraError>;
+    async fn update_status(&mut self, indexer: UpdateIndexerStatusDb) -> Result<IndexerModel, InfraError>;
+    async fn update_status_and_process_id(
```

Comment:
these functions shouldn't be part of the `Repository` trait as they are specific to `IndexerRepository`

Reply:
Hey, just unresolving this one as I still have the doubt, shouldn't `update_status` and `update_status_and_process_id` not be a part of the generic `Repository` trait

Reply:
#28